pyccel/parser/semantic.py

```diff
@@ -2632,7 +2632,10 @@ def _visit_Assign(self, expr, **settings):
             new_expr = Assign(l, r)
 
             if is_pointer_i:
-                new_expr = AliasAssign(l, r)
+                if isinstance(expr, AugAssign):
+                    new_expr = AugAssign(l, expr.op, r)
+                else:
+                    new_expr = AliasAssign(l, r)
 
             elif isinstance(expr, AugAssign):
                 new_expr = AugAssign(l, expr.op, r)
```

Comment:
Why do you duplicate this line of code:
```python
new_expr = AugAssign(l, expr.op, r)
```
rather than re-ordering the `if`?
```python
if isinstance(expr, AugAssign):
    new_expr = AugAssign(l, expr.op, r)
elif is_pointer_i:
    new_expr = AliasAssign(l, r)
```

Reply:
fixed https://github.com/pyccel/pyccel/pull/1286/commits/0345a8d7484f597364948d4d65eff6a70405ff2a https://github.com/pyccel/pyccel/pull/1286/commits/f8459aa46b3f9da260c48f8c45604567712663ad